independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -2179,10 +2179,13 @@ protected void implementIsAlternative(BeanInfo bean, ClassCreator beanCreator) {
 
     protected void implementGetPriority(BeanInfo bean, ClassCreator beanCreator) {
         if (bean.getPriority() != null) {
+            MethodCreator hasPriority = beanCreator.getMethodCreator("hasPriority", boolean.class)
+                    .setModifiers(ACC_PUBLIC);
+            hasPriority.returnBoolean(true);
+
             MethodCreator getPriority = beanCreator.getMethodCreator("getPriority", int.class)
                     .setModifiers(ACC_PUBLIC);
-            getPriority
-                    .returnValue(getPriority.load(bean.getPriority()));
```

Comment:
Why do we no longer need the `getPriority.load(...)`?

Reply:
That happens behind the scenes, `returnInt()` is a shortcut.